src/probnum/filtsmooth/gaussfiltsmooth/kalmanposterior.py

```diff
@@ -122,3 +121,31 @@ def __len__(self):
 
     def __getitem__(self, idx):
         return self.state_rvs[idx]
+
+    def sample(self, locations=None, size=()):
+        """
+        Draw samples from the filtering/smoothing posterior.
+
+
+        If location is None, it returns N joint samples from the list of RVs;
+        if location is one of the locations of the posterior,
+        if not (i.e. if it is a scalar), it returns a sample from the specified location.
+
+        If size is empty, it is a single sample. If not, multiple samples at once.
```

Comment:
I think it could also be helpful to describe the output some more, especially the shape.